android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.remotesettings
+
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.appservices.remotesettings.RemoteSettings
+import mozilla.appservices.remotesettings.RemoteSettingsConfig
+import mozilla.appservices.remotesettings.RemoteSettingsException
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileReader
+import java.io.FileWriter
+import java.io.IOException
+
+/**
+ * Helper class to download collections from remote settings in app services.
+ *
+ * @property storageRootDirectory The top level app-local storage directory.
+ * @property serverUrl The optional url for the settings server. If not specified, the standard server will be used.
+ * @property bucketName The optional name of the bucket containing the collection on the server.
+ * If not specified, the standard bucket will be used.
+ * @property collectionName The name of the collection for the settings server.
+ */
+class RemoteSettingsClient(
+    private val storageRootDirectory: File,
+    private val serverUrl: String? = "https://firefox.settings.services.mozilla.com",
+    var bucketName: String? = "main",
+    var collectionName: String = "",
+) {
+
+    private val config = RemoteSettingsConfig(
+        serverUrl = serverUrl,
+        bucketName = bucketName,
+        collectionName = collectionName,
+    )
+
+    /**
+     * Fetches all records for a collection that can be found in the local storage.
+     **/
+    suspend fun fetch(): JSONObject? = withContext(Dispatchers.IO) {
+        val fileName = "list.json"
```

Comment:
I think we'll probably want to figure out a robust strategy for saving different collections to different files. We may be able to do something as simple as prepending the collection name to the file, but I'm not sure if we'll need to worry about naming collisions. Can two collections on a server share a name? Two collections in a bucket?